scripts/tests/test-ci-all.sh

```diff
@@ -151,6 +146,64 @@ function always_success_test() {
 }
 export -f always_success_test
 
+tagged_versions=("$@")
+num_versions="$#"
+versions=( "current" "${tagged_versions[@]}" )
+if [[ "$num_versions" == "0" ]]; then
+  current_binaries_count=3
+else
+  # precompile binaries
+  binaries=( "fedimintd" "fedimint-cli" "gateway-cli" "gatewayd" )
+  parallel nix_build_binary_for_version "{1}" "{2}" ::: "${binaries[@]}" ::: "${tagged_versions[@]}"
```

Comment:
Is running the build in parallel actually useful? At which point will resource contention slow down things again? Significant parts of the build process run at 100% CPU for me. @dpc

Reply:
I'm not sure if running this with `parallel` will have any performance gains relative to nested for loops, but I was a fan of how concise parallel allows us to write nested for loops.

Worth noting that precompiling here does have a benefit later since calling `run_test_for_versions` in parallel will kick off multiple nix builds for the same binary and version (https://github.com/fedimint/fedimint/pull/4413/files#diff-6f67d3a8330a3c924847c4e3e44fa893ed4178fee92b9bfeba68fd6fdbaff5ecR43-R45).

Reply:
> Significant parts of the build process run at 100% CPU for me.

Are you running at least `CARGO_PROFILE=ci` build profile?